composeApp/src/commonMain/kotlin/presentation/screens/feed/FeedViewModel.kt

```diff
@@ -0,0 +1,41 @@
+package presentation.screens.feed
+
+import ShareManager
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import presentation.screens.feed.data.FeedRepository
+import presentation.screens.feed.model.Post
+
+data class FeedUiState(
+    val posts: List<Post> = emptyList(),
```

Comment:
Can we add the immutable list to avoid unnecessary recompositions?

I am not sure if we already have this dependency in the project.

```suggestion
    val posts: ImmutableList<Post> = emptyList(),
```

`ImmutableList` is a `Stable` type of `List` 🤗